### PR TITLE
[actions] Create new GitHub actions: build, release and publish

### PR DESCRIPTION
**Pull request by jjmerchante**
This pull request includes the following GitHub actions grouped by directory:
- build: build a Python package using Poetry and store the package as an artifact for future jobs.
- release: create a new GitHub release for the repository using the tag that triggered the workflow
- publish: publish a Python package to PyPI using Poetry

Related to https://github.com/chaoss/grimoirelab/issues/462

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# GrimoireLab GitHub actions
+
+# build
```
Comment (sduenas) [inline README.md]:
```suggestion
## build
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+```yaml
+steps:
+  - uses: chaoss/grimoirelab-github-actions/build@master
+    with:
+      artifact-name: grimoire-dist
+      artifact-path: dist
+```
+
+
+# release
```
Comment (sduenas) [inline README.md]:
```suggestion
## release
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+Create a new GitHub release for the repository using the tag that triggered the workflow and the notes available under `releases/<tag>.md`.
```
Comment (sduenas) [inline README.md]: This line is too long. Please split it in several ones.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+### Usage
+```yaml
+steps:
+  - uses: chaoss/grimoirelab-github-actions/release@master
+    with:
+      github-token: ${{ secrets.GITHUB_TOKEN }}
+```
+
+
+# publish
```
Comment (sduenas) [inline README.md]:
```suggestion
## publish
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+Publish a Python package to PyPI using Poetry. The package should be already built, you can use `grimoirelab-github-actions/build`.
```
Comment (sduenas) [inline README.md]: This line is too long too.

File: build/action.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Install poetry
+      run: |
+        curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -
```
Comment (sduenas) [inline build/action.yml]: [According to the documentation](https://python-poetry.org/docs/#osx--linux--bashonwindows-install-instructions), this has been changed to `get-poetry.py`.

File: publish/action.yml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        path: ${{ inputs.artifact-path }}
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Install poetry
+      run: |
+        curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -
```
Comment (sduenas) [inline publish/action.yml]: Here you should change it too.

**Review by sduenas — CHANGES_REQUESTED** [review]
In general it looks good to me. I think we should update how poetry is installed (according to their documentation).

File: build/action.yml (hunk 6 above)
Comment (jjmerchante) [reply build/action.yml]: The documentation says that `get-poetry.py` will be replaced by `install-poetry.py`


> The get-poetry.py script described [here](https://python-poetry.org/docs/master/#installation) will be replaced in Poetry 1.2 by install-poetry.py. From Poetry 1.1.7 onwards, you can already use this script as described here.

In the master branch they recommend installing poetry using the following command, so maybe is the best approach:
`curl -sSL https://install.python-poetry.org | python3 -`

Comment (sduenas) [reply build/action.yml]: My mistake!

**Review by sduenas — APPROVED** [review]
LGTM

**Pull request merged**